src/API.jl

```diff
@@ -827,9 +827,17 @@ function instantiate(ctx::Context; manifest::Union{Bool, Nothing}=nothing,
     # check if all source code and artifacts are downloaded to exit early
     Operations.is_instantiated(ctx) && return
 
-    Types.update_registries(ctx)
     pkgs = Operations.load_all_deps(ctx)
-    Operations.check_registered(ctx, pkgs)
+    try
+        # First try without updating the registry
+        Operations.check_registered(ctx, pkgs)
+    catch e
+        if !(e isa PkgError) || update_registries == false
```

Comment:
What is `update_registries` supposed to refer to? With Julia 1.5.1 you can get the error
```
ERROR: LoadError: UndefVarError: update_registries not defined
Stacktrace:
 [1] instantiate(::Pkg.Types.Context; manifest::Nothing, update_registry::Bool, verbose::Bool, platform::Pkg.BinaryPlatforms.Linux, kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Pkg/src/API.jl:836
```
before getting to
```
expected package `XXXXX [xxxxxxxxx]` to be registered
```

Reply:
My bad: https://github.com/JuliaLang/Pkg.jl/pull/1988